### PR TITLE
fix: Message string when someone's identity changed and user name is not available 

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This PR introduces improvements to the message handling functionality, particularly in scenarios where the user name is unavailable & user's identity has changed. A placeholder is now added for better user experience and readability. Additionally, a new string is provided for these cases to ensure clarity in communication. 

# Changes

1. **User Name Placeholder**: Implemented a placeholder for cases where the user name is not available. This change ensures that messages remain clear and understandable even when specific user details are missing.

2. **New String for Unavailable User Name**: Introduced a new localized string to be used when the user name is not available. This string is designed to be user-friendly and aligns with our application's localization standards.

3. **Snapshot Test Case**: Added a new snapshot test to cover the scenario where the user name is unavailable. This test ensures that the new placeholder and string are rendered correctly and function as expected.

# Testing

The new snapshot test has been added to our test suite and passed successfully.

# Note

We should avoid construct localized strings in code and instead have different localized strings for each case. I'm planning to tackle this problem on a separate PR. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS Tests/ReferenceImages/ConversationSystemMessageTests/testDecryptionFailedIdentifyChanged_ThereIsNoUser.320-0.png
Comment (johnxnguyen) [inline wire-ios/Wire-iOS Tests/ReferenceImages/ConversationSystemMessageTests/testDecryptionFailedIdentifyChanged_ThereIsNoUser.320-0.png:1]: This doesn't really make grammatical sense. Wouldn't it be better to say "A user's device identity changed", or "Someone's device identity changed"?

**Review by johnxnguyen — COMMENTED** [review]
What did it say before when the name was unavailable?

**Comment by agisilaos** [comment]
> What did it say before when the name was unavailable?

@johnxnguyen Nothing. Please take a look:

Screenshot 2023-11-27 at 15 44 09

**Comment by agisilaos** [comment]
Closing this for now since I didn't get an update from product/design what the content of the string should be.

**Pull request closed without merging**